### ISSUE TITLE
最后一个epoch验证不完，以及shape问题

### ISSUE THREAD

**Issue by sdreamforchen** [PaddlePaddle/PaddleDetection#9261]
### 问题确认 Search before asking

- [X] 我已经查询[历史issue](https://github.com/PaddlePaddle/PaddleDetection/issues)，没有发现相似的bug。I have searched the [issues](https://github.com/PaddlePaddle/PaddleDetection/issues) and found no similar bug report.

### Bug描述 Describe the Bug

您好
   1 在训练RT-DETR的时候，老是最后的eval，只eval到1000就不动了，每次都需要手动eval 最后的权重，有点麻烦。
   2 在rtdetr_transformer.py里面使用x=self.lamda * F.softmax(x) + (1 - self.lamda) * F.sigmoid(x)，在169行和170行的attention_weights = F.sigmoid(attention_weights).reshape([bs, Len_q, self.num_heads, self.num_levels, self.num_points])会报 shape错误。完整代码是
class Act(nn.Layer):
    def __init__(self, step=30000, r0=1.0):
        super(Act, self).__init__()
        self.iter = step+1
        self.total_step=step
        self.r0 = r0
        
        self.lamda = paddle.full([1],1,"float32")
        
    def forward(self, x):
        
        if self.training:
            
            if self.iter > 0:
                self.iter=self.iter - 1
            #    a = self.iter
            
            
            #print(self.iter)
            
            x1 = F.softmax(x)
            x2 = F.sigmoid(x)
                                     
            if self.iter<=self.total_step:
                self.lamda = paddle.to_tensor(self.r0*self.iter/self.total_step)
            else:
                self.lamda = paddle.to_tensor(self.r0)
            x = paddle.to_tensor(self.lamda * x1 + (1 - self.lamda) * x2)
        else:
            #x = self.norm2(x)
            if self.iter==0:
                x = paddle.to_tensor(F.sigmoid(x))
            else:
                x = paddle.to_tensor(self.lamda * F.softmax(x) + (1 - self.lamda) * F.sigmoid(x))
        #mm=self.lamda
        return x

class PPMSDeformableAttention(MSDeformableAttention):
         ........
        attention_weights = Act(attention_weights).reshape([bs, Len_q, self.num_heads, self.num_levels, self.num_points])
        .......

### 复现环境 Environment

最新版

### Bug描述确认 Bug description confirmation

- [X] 我确认已经提供了Bug复现步骤、代码改动说明、以及环境信息，确认问题是可以复现的。I confirm that the bug replication steps, code change instructions, and environment information have been provided, and the problem can be reproduced.

### 是否愿意提交PR？ Are you willing to submit a PR?

- [X] 我愿意提交PR！I'd like to help by submitting a PR!

**Comment by leo-q8** [comment]
是自己的数据吗，paddle 版本是多少呢？

**Comment by TingquanGao** [comment]
The issue has no response for a long time and will be closed. You can reopen or new another issue if are still confused.

---
_From Bot_
